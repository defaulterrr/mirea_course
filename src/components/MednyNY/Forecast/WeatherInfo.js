import React from 'react';
import PropTypes from 'prop-types';
import styles from './weather.module.css';

class WeatherInfo extends React.Component {
  constructor() {
    super();
    this.state = {
      weatherData: null,
    };
  }

  componentDidMount() {
    const { zip } = this.props;
    const apiKey = 'b1b35bba8b434a28a0be2a3e1071ae5b';
    const URL = `http://api.openweathermap.org/data/2.5/weather?q=${zip}&appid=${apiKey}&units=metric`;
    fetch(URL).then((res) => res.json()).then((json) => {
      this.setState({ weatherData: json });
    });
  }

  render() {
    const { weatherData } = this.state;
    if (!weatherData) { return <div>Loading</div>; }
    const weather = weatherData.weather[0];
    return (
      <div className={styles.mainDiv}>
        <form className={styles.form}>
          <div className={styles.img}>
            <img className={styles.weatherImage} alt="weather" src="https://i.pinimg.com/564x/1a/5e/15/1a5e153490c4fd46cb8d3a8254dbba48.jpg" />
          </div>
          <div className={styles.info}>
            <h1 className={styles.title}>
              {weather.main}
              {' '}
              in
              {' '}
              {weatherData.name}
            </h1>
            <p className={styles.parag}>
              Temprature:
              {(weatherData.main.temp).toFixed(0)}
              °С
            </p>
            <p className={styles.parag}>
              High Temp:
              {(weatherData.main.temp_max).toFixed(0)}
              °С
            </p>
            <p className={styles.parag}>
              Low Temp:
              {(weatherData.main.temp_min).toFixed(0)}
              °С
            </p>
            <p className={styles.parag}>
              Wind Speed:
              {(weatherData.wind.speed).toFixed(0)}
              {' '}
              mi/hr
            </p>
            <p className={styles.parag}>
              Humidity:
              {weatherData.main.humidity}
            </p>
          </div>
        </form>
      </div>
    );
  }
}

WeatherInfo.propTypes = {
  zip: PropTypes.string.isRequired,
};

export default WeatherInfo;
