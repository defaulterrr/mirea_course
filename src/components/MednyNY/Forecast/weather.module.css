@import url(https://fonts.googleapis.com/css?family=Lato:100,300,400);
.mainDiv {
    margin-top: 10px;
    font-family: 'Lato', sans-serif;
}

.form{
    width: 650px;
    height: 400px;
    background-color: rgb(172, 166, 166);
    border-radius: 25px;
}

.img {
    float: left;
    width: 200px;
    height: 300px;
}

.weatherImage{
    width: 300px;
    height: 400px;
    border-radius: 25px;
}

.info{
    float: right;
    margin-right: 70px;
    width: 200px;
    height: 400px;
}

.title{
    float: left;
    padding-bottom: 20px;
    font-size: 25px;
    width: 200px;
}

.parag{
    width: 200px;
    text-align: left;
    padding-left: -10px;
}

.button {
    color: white;
    background-color: rgb(172, 166, 166);
    width: 100px;
    height: 40px;
    border-radius: 25px;
    border: none;
    cursor: pointer;
}