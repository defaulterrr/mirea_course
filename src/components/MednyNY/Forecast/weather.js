import React from 'react';
import WeatherInfo from './WeatherInfo';
import styles from './weather.module.css';

class Forecast extends React.Component {
  PLACES = [
    { name: 'Moscow', zip: '101000' },
    { name: 'Krasnoyarsk', zip: '660000' },
  ];

  constructor() {
    super();
    this.state = {
      activePlace: 0,
    };
  }

  render() {
    const { activePlace } = this.state;
    return (
      <div className="App">
        {this.PLACES.map((place, index) => (
          <button
            type="button"
            className={styles.button}
            key={index}
            onClick={() => {
              this.setState({ activePlace: index });
            }}
          >
            {place.name}
          </button>
        ))}
        <WeatherInfo
          key={activePlace}
          zip={this.PLACES[activePlace].zip}
        />
      </div>
    );
  }
}

export default Forecast;
