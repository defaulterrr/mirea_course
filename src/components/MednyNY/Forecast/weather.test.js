import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Forecast from './weather';

configure({ adapter: new Adapter() });
describe('work correctly', () => {
  it('match snapshout', () => {
    const WeatherComponent = renderer.create(<Forecast />).toJSON();
    expect(WeatherComponent).toMatchSnapshot();
  });

  it('Should find buttons', () => {
    const wrapper = shallow(<Forecast />);
    const container = wrapper.find('button');
    expect(container.length).toEqual(2);
  });
});
