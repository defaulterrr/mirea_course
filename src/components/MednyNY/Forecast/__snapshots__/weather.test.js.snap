// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`work correctly match snapshout 1`] = `
<div
  className="App"
>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Moscow
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Krasnoyarsk
  </button>
  <div>
    Loading
  </div>
</div>
`;
